scene/2d/navigation_agent_2d.cpp

```diff
@@ -274,7 +274,6 @@ void NavigationAgent2D::_notification(int p_what) {
 						NavigationServer2D::get_singleton()->agent_set_velocity_forced(agent, velocity_forced);
 					}
 				}
-				_check_distance_to_target();
```

Comment:
Since reaching the target now ends the navigation, it's not enough anymore to just try to update `target_reached` and emit the corresponding signal on each frame. We need to also update the navigation state and I think this should only be done in `_update_navigation`.